src/compiler/scanner.ts

```diff
@@ -1341,6 +1341,19 @@ namespace ts {
             return -1;
         }
 
+
+        function peekExtendedUnicodeEscape(): number {

```

Comment:
Note that this just peeks to make sure the escape scans and doesn't validate that the resulting number is in the expected range - `scanExtendedUnicodeEscape` (which is called after both uses to get the real value) does all that.